capella2polarion/converters/link_converter.py

```diff
@@ -225,13 +223,13 @@ def create_grouped_link_fields(
                     key = link.secondary_work_item_id
                     back_links.setdefault(key, []).append(link)
 
-            role_id = self._remove_prefix(role)
             config: converter_config.LinkConfig | None = None
             for link_config in data.type_config.links:
-                if link_config.polarion_role == role_id:
+                if link_config.polarion_role == role:
                     config = link_config
                     break
 
+            role_id = self._remove_prefix(role)
```

Comment:
If we want to go for the approach of having the prefix logic in the Config, we should do it completely. The prefix should not be part of the LinkConverter anymore and we should offer in the `LinkConfig` not only a `role` but also a `link_field`. This custom field can be defined in the config individually and if it was not configured individually it will be filled with the `polarion_role` without any prefix. This way, we could completely rely on the `link_field` parameter here and don't have to care about any prefixes anymore.
For the reverse fields we would use the `reverse_field` attribute so that there is no need to pass the `role` to `_create_link_fields` as all information are available in the `LinkConfig`.

Reply:
This also means that the `config` parameter will be mandatory to call `_create_link_fields`, but this should not be an issue - we just have to make sure that we also respect this for `description_references` and `diagram_elements`